recipes-support/provisioning/files/phytec-board-config.sh

```diff
@@ -0,0 +1,403 @@
+            valid=$(echo ${response} | jq .valid_until)
+            whiptail --msgbox "\
+Welcome to the ESEC IoT Device Manager Platform
+The next steps are:
+ 1) Login to your account on the ESEC IoT Device
+    Manager Platform https://iot.aws.esec-experts.com.
+    If you do not have an account, then use
+    option 1) New Account and Onboarding to register
+ 2) Add the
+    Token: ${token}
+    Valid until: ${valid}
+    for the device with
+    Certificate serial number ${certserial}
+    to your account:" $WT_HEIGHT $WT_WIDTH
+        else
+            # get user item
+            set_onboarded accountdevice
+            check=$(echo ${response} | jq .user_item.customer_name)
+            whiptail --msgbox "\
+Welcome ${check}
+to the ESEC IoT Device Manager Platform
+The next steps are:
+ 1) Check your email account for a message
+    from welcome@esec-experts.com
+ 2) Verify your email
+ 3) Login to your IoT Device Manager account
+ 4) Check the state of your device in
+    your IoT Device Manager account" $WT_HEIGHT $WT_WIDTH
+        fi
+    else
+        if [ $# -eq 0 ]; then
+            set_onboarded tokendevice
+        else
+            set_onboarded accountdevice
+        fi
+        echo "response: ${response}"
+    fi
+    return 0
+}
+
+do_newaccount() {
+    do_contract
+    if [ "${ESECCONTRACT}" = False ]; then
+        return 0
+    fi
+    account=$(whiptail --inputbox "Your email address for the new account:" $WT_HEIGHT $WT_WIDTH 3>&1 1>&2 2>&3)
+    if [ $? -ne 0 ]; then
+        return 1
+    elif [ -z "$account" ]; then
+        whiptail --msgbox "Error: your email address is empty!: $account. " $WT_HEIGHT $WT_WIDTH
+        return 1
+    fi
+    do_onboarding $account
+    return $val
+}
+
+
+calc_wt_size
+do_esecawsconf
+#
+# Command line options for non-interactive use
+#
+for i in $*
+do
+    case $i in
+    -o|--onboarding)
+        do_onboarding
+        exit 0
+        ;;
+    --newaccount=*)
+        OPT=`echo $i | sed 's/[-a-zA-Z0-9]*=//'`
+        do_onboarding $OPT
+        exit 0
+        ;;
+    -n)
+        INTERACTIVE=True
+        do_newaccount
+        exit 0
+        ;;
+    acceptcontract)
+        set_eseccontract
+        INTERACTIVE=False
+        ;;
+    -h|--help)
+        echo "$usage"
+        exit 0
+        ;;
+    *)
+        # unknown option
+        ;;
+    esac
+done
+
+#
+# Interactive use loop
+#
+if [ "$INTERACTIVE" = True ]; then
+    while true; do
+        FUN=$(whiptail --title "PHYTEC - ESEC IoT Device Configuration Tool" --backtitle "$(cat /proc/device-tree/model)" --menu "Setup Options" $WT_HEIGHT $WT_WIDTH $WT_MENU_HEIGHT --cancel-button Finish --ok-button Select \
+            "1 New Account and Onboarding" "for ESEC IoT Device Manager Platform" \
+            "2 Onboarding" "for ESEC IoT Device Manager Platform" \
+            "3 Contract" "for ESEC IoT Device Manager Platform" \
+            "4 About" "ESEC IoT Device Manager Platform" \
+            3>&1 1>&2 2>&3)
```

Comment:
I get the following for this line:
```
/usr/bin/phytec-board-config: line 388: warning: command substitution: ignored null byte in input
```
The contents of `/proc/device-tree/model` contains NULL characters and thus we get this warning. To suppress this use `$(tr -d '\0' < /proc/device-tree/model)`, which truncates the NULL characters from the device-tree model string (we do not need those anyway).

Reply:
is changed